data_model/src/sorting.rs

```diff
@@ -0,0 +1,68 @@
+//! Structures and traits related to sorting.
+
+#[cfg(not(feature = "std"))]
+use alloc::{
+    borrow::ToOwned as _,
+    collections::btree_map,
+    format,
+    string::{String, ToString as _},
+    vec,
+    vec::Vec,
+};
+#[cfg(feature = "std")]
+use std::collections::btree_map;
+
+use serde::{Deserialize, Serialize};
+#[cfg(feature = "warp")]
+use warp::{Filter, Rejection};
+
+use crate::prelude::*;
+
+const SORT_BY_KEY: &str = "sort_by_key";
+
+/// Structure for sorting requests
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct Sorting {
+    /// [`Name`] of the key in [`Asset`]'s metadata used to order query result.
+    pub sort_by_key: Option<Name>,
```

Comment:
I don't think this should be optional. If additional sorting possibilities are introduced, this should be turned into an enum. Maybe you should make it enum even now

Reply:
also, I think `sort_by_metadata_key` would be more appropriate naming to make the intended use more clear

Reply:
you could do it like:
```rs
pub struct Sorting {
    NoSorting,
    ByKey(Name),
}
```